csharp/BYOAuthFlow/Program.cs

```diff
@@ -0,0 +1,228 @@
+        {
+            Console.WriteLine("Starting Cloud Scenario");
+            
+            var platformCredentials = new PlatformRefreshTokenCredential(RefreshToken);
+            RunScenario(
+                "locator.improbable.io",
+                443,
+                PlayerIdentityTokenServiceClient.Create(credentials: platformCredentials),
+                LoginTokenServiceClient.Create(credentials: platformCredentials),
+                DeploymentServiceClient.Create(credentials: platformCredentials),
+                insecureConnection: false);
+
+            Console.WriteLine("Starting Spatiald Scenario");
+            var spatialdEndpoint = new PlatformApiEndpoint("localhost", SpatialdPort, true);
+            RunScenario(
+                "localhost",
+                SpatialdPort,
+                PlayerIdentityTokenServiceClient.Create(spatialdEndpoint),
+                LoginTokenServiceClient.Create(spatialdEndpoint),
+                DeploymentServiceClient.Create(spatialdEndpoint),
+                insecureConnection: true);
+        }
+
+        private static void RunScenario(string locatorHost, ushort locatorPort, PlayerIdentityTokenServiceClient pitClient, LoginTokenServiceClient ltClient,
+            DeploymentServiceClient dsClient, bool insecureConnection)
+        {
+            var deployment = Setup(dsClient);
+            try
+            {
+                var pit = CreatePIT(pitClient);
+                deployment = FindDeployment(dsClient, deployment.Id);
+                var loginToken = CreateLoginTokenForDeployment(ltClient, pit, deployment.Id);
+                ConnectToLocator(locatorHost, locatorPort, insecureConnection, pit, loginToken);
+            }
+            finally
+            {
+                Cleanup(dsClient, deployment);
+            }
+        }
+
+        private static string CreatePIT(PlayerIdentityTokenServiceClient pitClient)
+        {
+            Console.WriteLine("Generate a PIT");
+            var playerIdentityTokenResponse = pitClient.CreatePlayerIdentityToken(
+                new CreatePlayerIdentityTokenRequest
+                {
+                    Provider = "provider",
+                    PlayerIdentifier = "player_identifier",
+                    ProjectName = ProjectName,
+                });
+            return playerIdentityTokenResponse.PlayerIdentityToken;
+        }
+
+        private static Deployment FindDeployment(DeploymentServiceClient dsc, string deploymentId)
+        {
+            Console.WriteLine("Choosing deployment");
+            return dsc.ListDeployments(new ListDeploymentsRequest
+            {
+                ProjectName = ProjectName
+            }).First(d => d.Id == deploymentId);
+        }
+
+        private static string CreateLoginTokenForDeployment(LoginTokenServiceClient ltClient,
+            string pit, string deploymentId)
+        {
+            Console.WriteLine("Generate a Login Token for the selected deployment");
```

Comment:
-> login token

Reply:
changed to LoginToken

Reply:
👍 (although looks like it's still as it was?)

Also, is there a reason this says "Generate" rather than "Generating"? It's the same for "Connect to the deployment..." and "Generate a PlayerIdentityToken" - should these be "Connecting" and "Generating" for consistency with the rest of the WriteLine outputs?

Reply:
Yep forgot that one!